### PR TITLE
chore: refactor custody component

### PR DESCRIPTION
**Pull request by zone-live**
## **Description**

Clean up and reorganize the MMI custody component to have a better readability and maintainability.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by albertolive — DISMISSED** [review]

File: ui/pages/institutional/manual-connect-custodian/manual-connect-custodian.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+              size={ButtonSize.Lg}
+              onClick={cancelConnectCustodianToken}
+            >
+              {t('cancel')}
+            </Button>
+            <Button
+              block
+              data-testid="jwt-form-connect-button"
+              size={ButtonSize.Lg}
+              onClick={async () => {
```
Comment (DDDDDanica) [inline ui/pages/institutional/manual-connect-custodian/manual-connect-custodian.tsx:142]: Nit: Can this onClick be moved out for cleaning code and easier reading purpose?

File: ui/pages/institutional/manual-connect-custodian/manual-connect-custodian.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      >
+        {window.innerWidth > 400 && (
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            marginBottom={4}
+            marginTop={4}
+          >
+            <ButtonIcon
+              data-testid="custody-back-button"
```
Comment (DDDDDanica) [inline ui/pages/institutional/manual-connect-custodian/manual-connect-custodian.tsx:88]: Nit: if this is not too much trouble, can this be refactored as `manual-connect-custodian-back-button` considering it's rendered in a lower level component?

File: ui/pages/institutional/custodian-list-view/custodian-list-view.test.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ];
+    const { container, getByText } = render(
+      <CustodianListView custodianList={mockCustodianList} />,
+    );
+    expect(container).toMatchSnapshot();
+
+    await waitFor(() => {
+      expect(getByText('Custodian 1')).toBeInTheDocument();
+      expect(getByText('Custodian 2')).toBeInTheDocument();
+    });
```
Comment (DDDDDanica) [inline ui/pages/institutional/custodian-list-view/custodian-list-view.test.tsx:25]: is `Custodian 3` assertion missed here?

File: ui/pages/institutional/custodian-accounts-connected/custodian-accounts-connected.test.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  it('should render CustodianAccountsConnected', async () => {
+    const { container, getByText } = render(<CustodianAccountsConnected />);
+    expect(container).toMatchSnapshot();
+
+    await waitFor(() => {
+      expect(
+        getByText('allCustodianAccountsConnectedTitle'),
+      ).toBeInTheDocument();
+    });
```
Comment (DDDDDanica) [inline ui/pages/institutional/custodian-accounts-connected/custodian-accounts-connected.test.tsx:29]: Nit: can we directly assert here?
```suggestion
     expect(
        getByText('allCustodianAccountsConnectedTitle'),
      ).toBeInTheDocument();
```

**Review by DDDDDanica — DISMISSED** [review]

**Review by albertolive — DISMISSED** [review]

**Review by albertolive — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**